saltgui/static/scripts/routes/grains.js

```diff
@@ -8,6 +8,28 @@ class GrainsRoute extends PageRoute {
 
     this._updateKeys = this._updateKeys.bind(this);
     this._updateMinion = this._updateMinion.bind(this);
+
+    // collect the list of displayed minions
+    let previewGrainsText = window.localStorage.getItem("preview_grains");
+    if(!previewGrainsText || previewGrainsText === "undefined") {
+      previewGrainsText = "[]";
+    }
+    this._previewGrains = JSON.parse(previewGrainsText);
+    if(!Array.isArray(this._previewGrains)) {
+      this._previewGrains = [ ];
+    }
+    // add all the required columns
+    const tr = document.querySelector("#page_grains thead tr");
+    while(tr.childElementCount > 6) {
+      tr.removeChild(tr.lastChild);
+    }
+    for(let i = 0; i < this._previewGrains.length; i++) {
+      const th = document.createElement("th");
+      th.innerText = this._previewGrains[i];
```

Comment:
It would be nice to have first letter in uppercase to be consistent with rest of the columns names. It can be easily done by this:
```
th.innerText = this._capitalizeFirstLetter(this._previewGrains[i]);
```

where new helper function looks like this:
```
  _capitalizeFirstLetter(string) {
    return string.charAt(0).toUpperCase() + string.slice(1);
  }
```

Reply:
I had thought about that and decided not to do that. I like that they look slightly different. It visually also explains why they are on the right hand side. If anything must be done later, I would prefer an explicit column label value in the configuration.

Reply:
Yes, but please let try to think a bit about motivation of the user who would like to promote some grains (as a special one) to top level. I don't think he would like to see them behaves differently? If you will take my as example I would like to promote `roles` and `environment` grains as a first class citizens inside grains table. If you still not convinced can you please at least consider configurable option where I could configure it. Please, please, please :)

Reply:
not in this PR. and also unlikely later. additional reason is that this also does not happen on grains-details page.